examples/nrf52/src/main.c

```diff
@@ -6,141 +6,29 @@
-                    0x90, 0x00,
-                };
-                assert(len >= sizeof(get_large_blob_response));
-                memcpy(buf, get_large_blob_response, sizeof(get_large_blob_response));
-                return sizeof(get_large_blob_response);
-            }
-    case FIDO_STATE_UNINIT:
-    default:
-        return 0;
-        break;
-    }
-}
-
-static int example_write(void *handle, const unsigned char *buf, const size_t len) {
-    // Output the buffer.
-    printk("writing: ");
-    for (size_t i = 0; i < len; ++i) {
-        printk("%02x ", buf[i]);
-    }
-    printk("\n");
-
-    // Stupid state machine, that does not know anything about parsing the message completely.
-    switch (sim_state) {
-        case FIDO_STATE_UNINIT:
-            sim_state = FIDO_STATE_APPLET_SELECTION;
-            break;
-        case FIDO_STATE_APPLET_SELECTION:
-            sim_state = FIDO_STATE_GET_INFO;
-            break;
-        case FIDO_STATE_GET_INFO:
-            sim_state = FIDO_STATE_GET_LARGE_BLOB;
-            break;
-        default: break;
-    }
-    return (int)len;
-}
-
-static const fido_dev_io_t nfc_io = {
-    .open = example_open,
-    .close = example_close,
-    .read = example_read,
-    .write = example_write
-};
+#include "stateless_rp/stateless_rp.h"
+#include "stateless_rp/stateless_rp_nfc_simulator.h"
+#include "hw_crypto.h"
 
 int main(void) {
-    fido_dev_t dev;
-    if (fido_init_nfc_device(&dev, &nfc_io) != FIDO_OK) {
-        return 1;
-    }
-
-    if (fido_dev_open(&dev) != FIDO_OK) {
-        return 2;
-    }
-
-    // Retrieve large blob.
-    uint8_t key[] = {
-        // 0xCA, 0x97, 0x81, 0x12, 0xCA, 0x1B, 0xBD, 0xCA, 0xFA, 0xC2, 0x31, 0xB3, 0x9A, 0x23, 0xDC, 0x4D, 0xA7, 0x86, 0xEF, 0xF8, 0x14, 0x7C, 0x4E, 0x72, 0xB9, 0x80, 0x77, 0x85, 0xAF, 0xEE, 0x48, 0xBB,
-        // from Chromium
-        0xF7, 0x8E, 0x65, 0x59, 0xF4, 0xE8, 0x70, 0xF2, 0xF0, 0x37, 0x41, 0x63, 0x85, 0x31, 0xEF, 0x31, 0x50, 0x8F, 0x76, 0x18, 0x73, 0x4B, 0x68, 0x7A, 0x4A, 0x42, 0x16, 0x65, 0xEA, 0x6A, 0x7F, 0xA2,
-    };
-    fido_blob_t blob;
-    uint8_t outbuf[2048] = {0};
-    fido_blob_reset(&blob, outbuf, sizeof(outbuf));
-    if (fido_dev_largeblob_get(&dev, key, sizeof(key), &blob) != FIDO_OK) {
-        return 3;
+    if (init_hw_crypto() != 0) {
+        printk("Failed to initialize cryptography. Aborting.\n");
+        return -1;
+    } else {
+        printk("Initialized cryptography.\n");
     }
 
-    if (fido_dev_close(&dev) != FIDO_OK) {
-        return 4;
+    fido_dev_t dev;
+    if (prepare_stateless_rp_nfc_simulator_device(&dev) != 0) {
+        printk("Could not setup simulator device.\n");
```

Comment:
Should we return here?

```suggestion
        printk("Could not setup simulator device.\n");
        return -1;
```